sphinx/builders/__init__.py

```diff
@@ -422,11 +432,10 @@ def read(self):
         Store all environment docnames in the canonical format (ie using SEP as
         a separator in place of os.path.sep).
         """
-        updated, reason = self.env.update_config(self.config, self.srcdir, self.doctreedir)
```

Comment:
So far, `env.config` is replaced by new one here. It means all features refer old config object on incremental build.
I think that is not intended behavior. So I move the replacement to more early stage.